src/taco-cli/cli/utils/vsEmulator.ts

```diff
@@ -0,0 +1,253 @@
+/**
+  *******************************************************
+  *                                                     *
+  *   Copyright (C) Microsoft. All rights reserved.     *
+  *                                                     *
+  *******************************************************
+  */
+
+/// <reference path="../../../typings/node.d.ts" />
+/// <reference path="../../../typings/Q.d.ts" />
+/// <reference path="../../../typings/validator.d.ts" />
+/// <reference path="../../../typings/tacoUtils.d.ts" />
+
+"use strict";
+
+import fs = require("fs");
+import os = require("os");
+import path = require("path");
+import child_process = require("child_process");
+import Q = require("q");
+import validator = require("validator");
+import resources = require("../../resources/resourceManager");
+import utils = require("taco-utils");
+import Logger = utils.Logger;
+
+/**
+ * Handles the Visual Studio Emulator for Android specific operations.
+ */
+export class VSEmulator {
+
+    private static CLI_PATH: string = process.env["ProgramFiles(x86)"] + "\\Microsoft Emulator Manager\\1.0\\emulatorcmd.exe";
+    private static LIST_COMMAND: string = "\"" + VSEmulator.CLI_PATH + "\"" + " list /sku:Android /type:device /state:installed";
+    private static DETAIL_COMMAND: string = "\"" + VSEmulator.CLI_PATH + "\"" + " detail /sku:Android /type:device /id:";
+    private static EMULATOR_PROFILE_REGEX: RegExp = /(.+)\s\|\s(.+)\s\|\s(.+)/g;
+    private static LOG_FILE: string = "VSAndroidEmulator.log";
+    private static INSTALLED_PROFILES: IEmulatorProfile[];
+
+    /**
+     * Checks if the VS Emulator is installed on the machine. 
+     */
+    public static isInstalled(): Boolean {
+        return fs.existsSync(path.resolve(VSEmulator.CLI_PATH));
+    }
+
+    /**
+     * Checks if the emulator is correctly set-up and has installed platforms.
+     */
+    public static isReady(): Q.Promise<boolean> {
+        if (!VSEmulator.isInstalled()) {
+            return Q.resolve(false);
+        } else {
```

Comment:
I agree it is not needed, but I feel the code is more readable if we keep it unchanged.